code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -293,3 +284,29 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 		spawn(10)
 			if(!src.stat)
 				horde()
+
+/mob/living/simple_animal/hostile/proc/living(var/atom/A)
+	if(isliving(A))

```

Comment:
These type checks should be unnecessary. You can just declare it as the wanted type in the parameter.